src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -248,24 +248,34 @@ private bool StartCore(ProcessStartInfo startInfo)
             {
                 Debug.Assert(stdinFd >= 0);
                 _standardInput = new StreamWriter(OpenStream(stdinFd, FileAccess.Write),
-                    new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), StreamBufferSize) { AutoFlush = true };
+                    GetEncoding(), StreamBufferSize) { AutoFlush = true };
             }
             if (startInfo.RedirectStandardOutput)
             {
                 Debug.Assert(stdoutFd >= 0);
                 _standardOutput = new StreamReader(OpenStream(stdoutFd, FileAccess.Read),
-                    startInfo.StandardOutputEncoding ?? Encoding.UTF8, true, StreamBufferSize);
+                    startInfo.StandardOutputEncoding ?? GetEncoding(), true, StreamBufferSize);
             }
             if (startInfo.RedirectStandardError)
             {
                 Debug.Assert(stderrFd >= 0);
                 _standardError = new StreamReader(OpenStream(stderrFd, FileAccess.Read),
-                    startInfo.StandardErrorEncoding ?? Encoding.UTF8, true, StreamBufferSize);
+                    startInfo.StandardErrorEncoding ?? GetEncoding(), true, StreamBufferSize);
             }
 
             return true;
         }
 
+        /// <summary>Creates an encoding from the current environment.</summary>
+        /// <returns>The encoding.</returns>
+        private static Encoding GetEncoding()
+        {
+            Encoding enc = EncodingHelper.GetEncodingFromCharset();
```

Comment:
IIRC, this is relatively expensive, as it wasn't intended to be used very frequently, but it's going to be used every time a process is created, and potentially multiple times for each process.  Could you take a look at the implementation, verify, and see if there's anything in it that could be cached?